analyzers/tests/SonarAnalyzer.Test/TestCases/BooleanLiteralUnnecessary.cs

```diff
@@ -228,7 +228,17 @@ class Repro7999CodeFixError
     {
         void Method(bool cond)
         {
-            if (cond == false) { } // Noncompliant, TP but code fix is wrong - it should be fixed to "if (!cond)"
+            if (cond == false) { }  // Noncompliant
+            if (cond != false) { }  // Noncompliant
+
+            if (cond == true) { }   // Noncompliant
+            if (cond != true) { }   // Noncompliant
+
+            if (false == cond) { }  // Noncompliant
+            if (false != cond) { }  // Noncompliant
+
+            if (true == cond) { }   // Noncompliant
+            if (true != cond) { }   // Noncompliant
```

Comment:
6/8 of these are not impacted, only `x == false` and `false == x` has special parsing.
I added all of them because they are best friends and like to hang out together (and I wanted to make sure I did not break them).